harness/features/variable.local.test.ts

```diff
@@ -45,37 +45,54 @@ const expectedVariablesByType = {
 }
 
 describe('Variable Tests - Local', () => {
+    // This helper method enumerates all the SDKs and calls each test suite
+    // for each SDK, and creates a test client from the name.
+    // All supported SDKs can be found under harness/types/sdks.ts
     forEachSDK((name) => {
         const expectedPlatform = getPlatformBySdkName(name, true)
 
+        // This describeCapability only runs if the SDK has the "local" capability.
+        // Capabilities are located under harness/types/capabilities and follow the same
+        // name mapping from the sdks.ts file under harness/types/sdks.ts
         describeCapability(name, Capabilities.local)(name, () => {
             const testClient = new LocalTestClient(name)
             let eventsUrl: string
 
             describe('initialized client', () => {
                 beforeAll(async () => {
-                    // mock endpoints
+                    // This allows us to mock out the our specific client (using the clientId),
+                    // allowing us to have multiple clients serving different clients without
+                    // conflicting. This one is used to mock the config that the local client is going to use
+                    // locally in all of its methods.
                     scope
                         .get(`/client/${testClient.clientId}/config/v1/server/${testClient.sdkKey}.json`)
                         .reply(200, config)
 
-                    // create client
-                    await testClient.createClient(true,{
+                    // Some tests require extra params for startup, so we can just create a new test client
+                    // for these specific tests.
+                    await testClient.createClient(true, {
```

Comment:
maybe explain the "waitForInitialized" thing happening here

Reply:
added a blurb